desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,402 @@
+import {
+    HoistModel,
+    LoadSpec,
+    managed,
+    PersistenceProvider,
+    PersistOptions,
+    PlainObject,
+    Thunkable,
+    XH
+} from '@xh/hoist/core';
+import {StoreRecordId} from '@xh/hoist/data/StoreRecord';
+import {action, bindable, computed, makeObservable, observable} from '@xh/hoist/mobx';
+import {executeIfFunction, pluralize} from '@xh/hoist/utils/js';
+import {capitalize, cloneDeep, isEqualWith, isNil, isString, sortBy, startCase} from 'lodash';
+import {ManageDialogModel} from './cmp/ManageDialogModel';
+import {SaveDialogModel} from './cmp/SaveDialogModel';
+import {runInAction} from 'mobx';
+import {PersistenceView, PersistenceViewTree} from '@xh/hoist/desktop/cmp/persistenceManager/Types';
+
+/**
+ * PersistenceManager provides re-usable loading, selection, and user management of named configs, which are modelled
+ * and persisted on the server as databased domain objects extending the `PersistedObject` trait.
+ *
+ * These generic configs are intended for specific use cases such as saved Grid views, Dashboards, and data import
+ * mapping configs. This model loads all available views from a configured endpoint and exposes a `provider` property
+ * that can be passed to any HoistModel that takes a `persistWith` config.
+ *
+ * Objects managed by this system can be private or shared based on a user's company and/or app roles. The `acl` field
+ * on each persisted object determines if and how it is shared. Users with the configured `adminRole` can share saved
+ * objects to all app users and save changes to shared objects directly. Users with the configured `editorRole` can
+ * share objects to their own company only and modify shared objects shared only within their company. All users can
+ * save/update/delete their own private objects.
+ */
+interface Entity {
+    /** Key used in JsonBlob */
+    name: string;
+    /** User-facing name/label for an object managed by this model. */
+    displayName?: string;
+}
+
+export interface PersistenceManagerConfig<T extends PlainObject> {
+    /** Entity name or object for this model. */
+    entity: string | Entity;
+    /** Whether user can publish or edit globally shared objects. */
+    canManageGlobal: Thunkable<boolean>;
+    /** Async callback triggered when view changes. Should be used to recreate the affected models. */
+    onChangeAsync: (value: T) => void;
+    /** Used to persist this model's selected ID. */
+    persistWith: PersistOptions;
+    /** Optional flag to allow empty view selection. Defaults false*/
+    enableDefault?: boolean;
+}
+
+export class PersistenceManagerModel<T extends PlainObject = PlainObject> extends HoistModel {
+    //------------------------
+    // Persistence Provider
+    // Pass this to models that implement `persistWith` to include their state in the view.
+    //------------------------
+    readonly provider: PersistOptions = {
```

Comment:
Wonder if we should make a new `PersistenceManagerProvider` type so the API is consistent with `DashViewProvider`.